### ISSUE TITLE
Intermittent OK (expected ERROR) in /workers/Worker_ErrorEvent_error.htm

### ISSUE THREAD

**Issue by emilio** [servo/servo#17460]
Seen in #17444 and #17443.

cc @asajeffrey

**Comment by asajeffrey** [comment]
Hmm, this is intermittent isn't it?

I think this may be a problem with the test (or possibly with the test harness). The test includes:
```js
  var worker = new Worker("support/ErrorEvent-error.js");
  worker.onerror = t1.step_func_done(function(e) {
    assert_true(/hello/.test(e.message));
    assert_equals(e.error, null);
  });
```
The problem is that `worker.onerror` doesn't call `e.preventDefault()`, so the event propagates up to the main script thread (this is correct by the spec https://html.spec.whatwg.org/multipage/workers.html#runtime-script-errors-2). This causes the test to fail. 

I think this is the expected behaviour of wptrunner: if a worker throws an exception, the test suite should report this as a failure. @jgraham may have an opinon?

So the fix is to correct the test, by adding `e.preventDefault()` to the error handler.

This may be the same root cause as #15668.

**Comment by emilio** [comment]
Fixed by #17475
